app/models/api_resource.rb

```diff
@@ -2,6 +2,10 @@ class ApiResource < ApplicationRecord
   include JsonbFieldsParsable
 
   after_initialize :inherit_properties_from_parent
+  
+  after_initialize do
```

Comment:
@Pralish 
Any specific reason for using `after_initialize` ? Because it gets called even for `Model.new`. We could use `after_find` which gets called only after a record has been fetched from the database.